src/battle_anim.c

```diff
@@ -0,0 +1,1852 @@
+            bankIdentity = 0;
+        else
+            bankIdentity = GetBankIdentity(bank);
+
+        sub_8118FBC(1, 0, 0, bankIdentity, unknownStruct.unk8, unknownStruct.unk0, unknownStruct.unk4, unknownStruct.unkA);
+
+        if (IsContest())
+            sub_80A46A0();
+    }
+    else
+    {
+        RequestDma3Fill(0, (void*)(VRAM + 0x6000), 0x2000, 1);
+        RequestDma3Fill(0, (void*)(VRAM + 0xF000), 0x1000, 1);
+        sub_80A6B90(&unknownStruct, 2);
+        CpuFill16(0, unknownStruct.unk0 + 0x1000, 0x1000);
+        CpuFill16(0, unknownStruct.unk4 + 0x400, 0x800);
+        SetAnimBgAttribute(2, BG_ANIM_PRIORITY, 2);
+        SetAnimBgAttribute(2, BG_ANIM_SCREEN_SIZE, 1);
+        SetAnimBgAttribute(2, BG_ANIM_AREA_OVERFLOW_MODE, 0);
+
+        bankSpriteId = gBankSpriteIds[bank];
+
+        gBattle_BG2_X =  -(gSprites[bankSpriteId].pos1.x + gSprites[bankSpriteId].pos2.x) + 0x20;
+        gBattle_BG2_Y =  -(gSprites[bankSpriteId].pos1.y + gSprites[bankSpriteId].pos2.y) + 0x20;
+
+        if (setSpriteInvisible)
+            gSprites[gBankSpriteIds[bank]].invisible = 1;
+
+        SetGpuReg(REG_OFFSET_BG2HOFS, gBattle_BG2_X);
+        SetGpuReg(REG_OFFSET_BG2VOFS, gBattle_BG2_Y);
+
+        LoadPalette(&gPlttBufferUnfaded[0x100 + bank * 16], 0x90, 0x20);
+        CpuCopy32(&gPlttBufferUnfaded[0x100 + bank * 16], (void*)(BG_PLTT + 0x120), 0x20);
+
+        sub_8118FBC(2, 0, 0, GetBankIdentity(bank), unknownStruct.unk8, unknownStruct.unk0 + 0x1000, unknownStruct.unk4 + 0x400, unknownStruct.unkA);
+    }
+}
+
+static void sub_80A46A0(void)
+{
+    s32 i, j;
+    struct UnknownAnimStruct2 unknownStruct;
+    u16 *ptr;
+
+    if (IsSpeciesNotUnown(gContestResources->field_18->field_0))
+    {
+        sub_80A6B30(&unknownStruct);
+        ptr = unknownStruct.unk4;
+        for (i = 0; i < 8; i++)
+        {
+            for (j = 0; j < 4; j++)
+            {
+                u16 temp = ptr[j + i * 32];
+
+                ptr[j + i * 32] = ptr[7 - j + i * 32];
+                ptr[7 - j + i * 32] = temp;
+            }
+        }
+        for (i = 0; i < 8; i++)
+        {
+            for (j = 0; j < 8; j++)
+                ptr[j + i * 32] ^= 0x400;
+        }
+    }
+}
+
+void sub_80A4720(u16 a, u16 *b, u32 c, u8 d)
+{
+    s32 i, j;
+    s32 var;
+
+    if (d == 0)
+        var = 32;
+    else
+        var = 64;
+    a <<= 12;
+    for (i = 0; i < var; i++)
+    {
+        for (j = 0; j < 32; j++)
+            b[j + i * 32] = ((b[j + i * 32] & 0xFFF) | a) + c;
+    }
+}
+
+void sub_80A477C(bool8 to_BG2)
+{
+    struct UnknownAnimStruct2 unknownStruct;
+    sub_80A6B30(&unknownStruct);
+
+    if (!to_BG2 || IsContest())
+    {
+        sub_80A6C68(1);
+        gBattle_BG1_X = 0;
+        gBattle_BG1_Y = 0;
+    }
+    else
+    {
+        sub_80A6C68(2);
+        gBattle_BG2_X = 0;
+        gBattle_BG2_Y = 0;
+    }
+}
+
+static void task_pA_ma0A_obj_to_bg_pal(u8 taskId)
+{
+    u8 r4;
+    u8 r6;
+    s16 r3;
+    s16 r2;
```

Comment:
You should name these vars more descriptively.

Reply:
copied from pokeruby, I know too little what the function does to give them better names

Reply:
Seems pretty clear to me that r3 is X, r2 is Y, r6 is palette index, and r4 is sprite index.